main.tf

```diff
@@ -361,3 +387,76 @@ resource "aws_s3_access_point" "ap" {
 
   depends_on = [var.module_depends_on]
 }
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Replication Role and policy
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_iam_role" "replication" {
+  count = local.replication_enabled ? 1 : 0
+  name  = "${var.bucket}-replication-role"
+
+  assume_role_policy = jsonencode(
+    {
+      "Version" : "2012-10-17",
+      "Statement" : [
+        {
+          "Action" : "sts:AssumeRole",
+          "Principal" : {
+            "Service" : "s3.amazonaws.com"
+          },
+          "Effect" : "Allow",
+          "Sid" : ""
+        }
+      ]
+    }
+  )
+}
+
+resource "aws_iam_policy" "replication" {
+  count = local.replication_enabled ? 1 : 0
+  name  = "${var.bucket}-replication-policy"
+
+  policy = jsonencode(
```

Comment:
please use `aws_iam_policy_document` data source instead of `jsonencode` for *all* policies as this fixes some AWS specific issues to prevent policies from constantly showing changes at plan time.